lib/cache/entry.js

```diff
@@ -194,6 +189,7 @@ class CacheEntry {
       this[_request] = new Request(this.entry.metadata.url, {
         method: 'GET',
         headers: this.entry.metadata.reqHeaders,
+        ...(this.entry.metadata.options || {}),
```

Comment:
Worth noting: the `|| {}` is necessary to handle existing caches with the new code since existing cache entries won't have a `metadata.options` stored. The result is that existing caches will assume previously cached requests were made with `{compress: true}` (because it's the default of `Request`) and match new requests also made with `{compress: true}`. If the previously cached requests were actually made with `{compress: false}`, then a stored compressed response may be returned to the new `{compress: true}` request. This is incorrect behaviour, but also the existing behaviour before the new code in this PR, so :man_shrugging:. The fix in this edge case is to delete the existing cache and let it repopulate.

AFAICT, there's no way to determine what the actual `compress` option was for these existing entries, so some value must be assumed. We could maybe instead propagate an "unknown" value forward into the `satisfies()` check, but all this would let us do is treat all existing cache entries as misses for new requests. It's not clear to me that's any better.

Reply:
since `this.entry.metadata` _does_ exist, it's safe to leave off the `|| {}` because `undefined` is allowed in an object spread, it just doesn't add anything.

```
> var entry = { metadata: {} }
> { ...entry.metadata.options }
{}
```

Reply:
Ah, didn't know that about the spread operator! I thought it would throw a TypeError or something. Will fix.